src/wifi.rs

```diff
@@ -1442,3 +1526,90 @@ impl WifiWait {
         }
     }
 }
+
+#[cfg(feature = "async")]
+pub struct AsyncWifiDriver<'d, T>
+where
+    T: BorrowMut<WifiDriver<'d>>,
+{
+    base_driver: T,
+    _p: PhantomData<WifiDriver<'d>>,
+}
+
+#[cfg(feature = "async")]
+impl<'d, T> AsyncWifiDriver<'d, T>
+where
+    T: BorrowMut<WifiDriver<'d>>,
+{
+    #[cfg(all(feature = "alloc", esp_idf_comp_nvs_flash_enabled))]
+    pub fn new(base_driver: T) -> Result<Self, EspError> {
+        Ok(AsyncWifiDriver {
+            base_driver,
+            _p: PhantomData,
+        })
+    }
+}
+
+#[cfg(all(feature = "async"))]
```

Comment:
Now THIS trait impl is the only thing you need to protect with a feature, but the feature should be `nightly` (as it needs nightly features), not `async`.